pandas/io/tests/test_common.py

```diff
@@ -139,3 +139,10 @@ def test_next(self):
             self.assertEqual(next_line.strip(), line.strip())
 
         self.assertRaises(StopIteration, next, wrapper)
+
+    def test_cparser_error_depr(self):
+        # see gh-12665
+        with tm.assert_produces_warning(DeprecationWarning,
+                                        check_stacklevel=False):
+            with tm.assertRaises(common.ParserError):
+                raise common.CParserError()
```

Comment:
@jorisvandenbossche: Actually, I get an error if you do that:
~~~python
try:
...
except CParserError:
...
TypeError: catching classes that do not inherit from BaseException is not allowed
~~~